File: solutions/go/flatten-array/1/flatten_array.go
package flatten

import "reflect"

func Flatten(nested interface{}) []interface{} {
	result := make([]interface{}, 0)
	for _, element := range nested.([]interface{}) {
		switch {
		case element == nil:
		case reflect.TypeOf(element).Kind() == reflect.Slice:
			result = append(result, Flatten(element)...)
		default:
			result = append(result, element)
		}
	}
	return result
}

File: solutions/typescript/flatten-array/1/flatten-array.ts
export function flatten(array: unknown[]): number[] {
  const flattened = []

  for (let i = 0; i < array.length; i++) {

    const element = array[i]
    
    if (typeof element === 'number')
      flattened.push(Number(element))
    if (Array.isArray(element))
      flattened.push(...flatten(element))
  }

  return flattened
}